services/api/src/service/comment.ts

```diff
@@ -287,3 +288,60 @@ export async function postNewComment({
     };
 
 }
+
+interface DeleteCommentBySlugIdProps {
+    db: PostgresJsDatabase;
+    commentSlugId: string;
+    userId: string;
+    authHeader: string;
+    didWrite: string;
+}
+
+export async function deleteCommentBySlugId({
+    db, commentSlugId, userId, authHeader, didWrite }: DeleteCommentBySlugIdProps): Promise<void> {
+
+    try {
+        await db.transaction(async (tx) => {
+            const updatedCommentIdResponse = await tx
+                .update(commentTable)
+                .set({
+                    currentContentId: null
+                })
+                .where(and(eq(commentTable.authorId, userId), eq(commentTable.slugId, commentSlugId)))
+                .returning({
+                    updateCommentId: commentTable.id,
+                    postId: commentTable.postId
+                });
+
+            if (updatedCommentIdResponse.length != 1) {
+                server.log.error("Invalid comment table update response length: " + updatedCommentIdResponse.length.toString());
```

Comment:
i think we should throw an error here instead of logging, probably after the tx.rollback(), though throwing would rollback anyway I think (to be verified)